contrib/isa-l-cmake/CMakeLists.txt

```diff
@@ -1,8 +1,5 @@
-option(ENABLE_ISAL_LIBRARY "Enable ISA-L library" ${ENABLE_LIBRARIES})
-if (ARCH_AARCH64)
-    # Disable ISA-L libray on aarch64.
-    set (ENABLE_ISAL_LIBRARY OFF)
-endif ()
+# ISA-L is geared towards x86-64, so it shall be disabled by default for other platforms
+option(ENABLE_ISAL_LIBRARY "Enable ISA-L library" $<AND:${ENABLE_LIBRARIES},$<STREQUAL:${CMAKE_SYSTEM_PROCESSOR},"x86_64">>)
```

Comment:
Do we need a "generator expression" here?

Reply:
It certainly isn't a necessity, but it's a bit more concise than other alternatives. If you'd prefer a simpler construct I could accommodate that as well.

Reply:
Let's change to a longer construction.

It's a coincidence that I know that this construction exists and even remembered how it's named, but for all normal people, it's pure obscurity.